app/scripts/controllers/bridge-status/bridge-status-controller.ts

```diff
@@ -0,0 +1,316 @@
+export type FetchBridgeTxStatusArgs = {
+  statusRequest: StatusRequest;
+};
+export default class BridgeStatusController extends StaticIntervalPollingController<
+  typeof BRIDGE_STATUS_CONTROLLER_NAME,
+  { bridgeStatusState: BridgeStatusControllerState },
+  BridgeStatusControllerMessenger
+> {
+  #pollingTokensBySrcTxHash: Record<SrcTxHash, string> = {};
+
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: BridgeStatusControllerMessenger;
+    state?: Partial<{
+      bridgeStatusState: BridgeStatusControllerState;
+    }>;
+  }) {
+    super({
+      name: BRIDGE_STATUS_CONTROLLER_NAME,
+      metadata,
+      messenger,
+      // Restore the persisted state
+      state: {
+        ...state,
+        bridgeStatusState: {
+          ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+          ...state?.bridgeStatusState,
+        },
+      },
+    });
+
+    // Register action handlers
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:startPollingForBridgeTxStatus`,
+      this.startPollingForBridgeTxStatus.bind(this),
+    );
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:wipeBridgeStatus`,
+      this.wipeBridgeStatus.bind(this),
+    );
+
+    // Set interval
+    this.setIntervalLength(REFRESH_INTERVAL_MS);
+
+    // If you close the extension, but keep the browser open, the polling continues
+    // If you close the browser, the polling stops
+    // Check for historyItems that do not have a status of complete and restart polling
+    this.restartPollingForIncompleteHistoryItems();
+  }
+
+  resetState = () => {
+    this.update((_state) => {
+      _state.bridgeStatusState = {
+        ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+      };
+    });
+  };
+
+  wipeBridgeStatus = ({
+    address,
+    ignoreNetwork,
+  }: {
+    address: string;
+    ignoreNetwork: boolean;
+  }) => {
+    // Wipe all networks for this address
+    if (ignoreNetwork) {
+      this.update((_state) => {
+        _state.bridgeStatusState = {
+          ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+        };
+      });
+    } else {
+      const { selectedNetworkClientId } = this.messagingSystem.call(
+        'NetworkController:getState',
+      );
+      const selectedNetworkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        selectedNetworkClientId,
+      );
+      const selectedChainId = selectedNetworkClient.configuration.chainId;
+
+      this.#wipeBridgeStatusByChainId(address, selectedChainId);
+    }
+  };
+
+  restartPollingForIncompleteHistoryItems = () => {
+    // Check for historyItems that do not have a status of complete and restart polling
+    const { bridgeStatusState } = this.state;
+    const historyItems = Object.values(bridgeStatusState.txHistory);
+    const incompleteHistoryItems = historyItems
+      .filter(
+        (historyItem) => historyItem.status.status !== StatusTypes.COMPLETE,
```

Comment:
Have you tested with FAILED transaction statuses? Those also need be treated as complete

Reply:
As discussed we will be handling FAILED bridges (also source tx succeeds, dest tx pending for a long time) in a follow up PR